src/renderer/component/viewers/threeViewer/index.jsx

```diff
@@ -18,7 +20,61 @@ type Props = {
   },
 };
 
-class ThreeViewer extends React.PureComponent<Props> {
+type State = {
+  error: ?string,
+  isReady: boolean,
+  isLoading: boolean,
+};
+
+class ThreeViewer extends React.PureComponent<Props, State> {
+  static testWebgl = new Promise((resolve, reject) => {
+    if (detectWebGL()) resolve();
+    else reject();
+  });
+
+  static createGeometry(data) {
+    const geometry = new THREE.Geometry();
+    geometry.fromBufferGeometry(data);
+    geometry.computeBoundingBox();
+    geometry.computeVertexNormals();
```

Comment:
Might want to add `geometry.mergeVertices();`

and if you end up having issues mapping textures, this might help:
```
function updateGeometryUv(geometry) {
  let max = geometry.boundingBox.max,
      min = geometry.boundingBox.min;
  let offset = new THREE.Vector2(0 - min.x, 0 - min.y);
  let range = new THREE.Vector2(max.x - min.x, max.y - min.y);
  let faces = geometry.faces;

  geometry.faceVertexUvs[0] = [];

  for(let i = 0; i < faces.length ; i++) {

      let v1 = geometry.vertices[faces[i].a],
          v2 = geometry.vertices[faces[i].b],
          v3 = geometry.vertices[faces[i].c];

      geometry.faceVertexUvs[0].push([
          new THREE.Vector2((v1.x + offset.x)/range.x ,(v1.y + offset.y)/range.y),
          new THREE.Vector2((v2.x + offset.x)/range.x ,(v2.y + offset.y)/range.y),
          new THREE.Vector2((v3.x + offset.x)/range.x ,(v3.y + offset.y)/range.y)
      ]);
  }
  geometry.uvsNeedUpdate = true;
}
```

Reply:
Also, some of the operations might be faster if you can do them before you convert the `GeometryBuffer` into a `Geometry`

Reply:
We don't really support textures ATM so that won't be a problem...